source/components/utils/EngineContext.tsx

```diff
@@ -11,7 +11,7 @@ export const EngineProvider = EngineContext.Provider
 
 type SituationProviderProps = {
 	children: React.ReactNode
-	situation: Partial<Record<DottedName, string | number | Object>>
+	situation: Partial<Record<DottedName, string | number | Record<string, any>>>
```

Comment:
Pourquoi ce changement ?

Reply:
pas le droit à `Object` type

Reply:
Hum ok, peut-être remplacer par `object` (avec un o minuscule) plutôt alors ?

Reply:
Bah en fait non on ne peut pas mettre `object` dans tous les cas, sinon tsc sort des erreurs "2536: Type 'Names' cannot be used to index type 'object'." lors d'un susbscripting du genre `nonApplicableMapping[parsedRule.dottedName]`.

J'ai l'impression que c'est lié au pb du `Object.entries`: il y a un truc un peu profond dans Typescript (pour l'instant) qui fait que les `object` sont figés au niveau de leurs properties.

A mon avis mieux vaut laisser comme ça pour l'instant.